src/Nancy/Json/JavaScriptSerializer.cs

```diff
@@ -29,459 +29,49 @@
 namespace Nancy.Json
 {
     using System;
-    using System.Collections;
     using System.Collections.Generic;
-    using System.ComponentModel;
     using System.IO;
-    using System.Linq;
-    using System.Reflection;
-    using System.Text;
-
-    using Nancy.Helpers;
+    using Nancy.Json.Simple;
 
     public class JavaScriptSerializer
     {
         internal const string SerializedTypeNameKey = "__type";
 
-        List<IEnumerable<JavaScriptConverter>> _converterList;
-        List<IEnumerable<JavaScriptPrimitiveConverter>> _primitiveConverterList;
-
-        int _maxJsonLength;
-        int _recursionLimit;
-        bool _retainCasing;
-        JavaScriptTypeResolver _typeResolver;
-        bool _iso8601DateFormat;
+        bool retainCasing;
 
-#if NET_3_5
-        internal static readonly JavaScriptSerializer DefaultSerializer = new JavaScriptSerializer(null, false, 2097152, 100);
-
-        public JavaScriptSerializer()
-            : this(null, false, 2097152, 100)
-        {
-        }
+        readonly NancySerializationStrategy serializerStrategy;
 
-        public JavaScriptSerializer(JavaScriptTypeResolver resolver)
-            : this(resolver, false, 2097152, 100)
-        {
-        }
-#else
-        internal static readonly JavaScriptSerializer DefaultSerializer = new JavaScriptSerializer(null, false, 102400, 100, false, true, null, null);
+        internal static readonly JavaScriptSerializer DefaultSerializer = new JavaScriptSerializer(false, JsonConfiguration.Default);
 
         public JavaScriptSerializer()
-            : this(null, false, 102400, 100, false, true, null, null)
+            : this(false,JsonConfiguration.Default)
         {
         }
 
-        public JavaScriptSerializer(JavaScriptTypeResolver resolver)
-            : this(resolver, false, 102400, 100, false, true, null, null)
+        public JavaScriptSerializer(bool registerConverters, JsonConfiguration configuration)

```

Comment:
It would make more sense if these parameters swapped order, and perhaps even an overload without the `registerConverters` parameter?